### PR TITLE
chore: added enhancements to dropdown to support icon and improved accessibilty

### PR DESCRIPTION
**Pull request by ayadav-eightfold**
added enhancements to dropdown to support icon and improved accessibilty

### CODE REVIEW

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]
Let make the changes as discussed on call

**Review by dkilgore-eightfold — APPROVED** [review]

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -121,7 +180,12 @@ export const Dropdown: FC<DropdownProps> = ({
         );
 
     return (
-        <div className={mainWrapperClasses} style={style}>
+        <div
+            className={mainWrapperClasses}
+            style={style}
+            ref={reference}
+            {...containerProps}
```
Comment (ychhabra-eightfold) [inline src/components/Dropdown/Dropdown.tsx:187]:
```suggestion
            {...TRIGGER_TO_HANDLER_MAP_ON_LEAVE[trigger] ? {TRIGGER_TO_HANDLER_MAP_ON_LEAVE[trigger]: toggle(false) : {}}}
```

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -70,6 +90,34 @@ export const Dropdown: FC<DropdownProps> = ({
         );
     }, [refs.reference, refs.floating, update]);
 
+    const onHoverElementFocus = useCallback((e) => {
+        if (
+            refs.reference.current instanceof Element &&
+            refs.reference.current?.contains(e.target)
```
Comment (ychhabra-eightfold) [inline src/components/Dropdown/Dropdown.tsx:96]:
```suggestion
            refs.reference.current?.contains?.(e.target)
```

File: src/hooks/useEventListener.ts [hunk 3]
```diff
@@ -42,7 +42,7 @@ export const useEventListener = (
 
     useEffect(() => {
         // Use window if no element is passed, otherwise default to null in SSR
-        const element = getElement(target);
+        const element = target && getElement(target);
```
Comment (ychhabra-eightfold) [inline src/hooks/useEventListener.ts:45]: is this needed ?

**Pull request merged**